src/Pirouette/SMT/Constraints.hs

```diff
@@ -60,14 +61,15 @@ data Branch lang meta = Branch
     newTerm :: TermMeta lang meta
   }
 
+-- TODO: Maybe this should be merged with 'LanguageSMT'
```

Comment:
Our implementation requires both instances to run the symbolic evaluation, so I guess it doesn't really matter, any `lang` must be `LanguageSMT lang` and `LanguageSMTBranches lang`.

I'm under the impression that these classes were separated because certain types used inside `LanguageSMTBranches` are not available where `LanguageSMT` is at and would require some puzzle solving to find non-cyclical imports to satisfy everything. I decided to leave it untouched for now.

Reply:
This might be a good moment to rename those classes. The general idea is:

```
LanguageBuiltins --> defines the built-ins (both terms and types)
  |     \
  |      LanguageBuiltinTypes --> defines the typing rules of each built-in term
  |
LanguageSMT --> defines translation of built-ins into SMT
  |             (not every term can be translated)
  |
LanguageSMTBranches --> defines at which points the symbolic evaluator has to do sth. special
```

on that sense, `LanguageSMTBranches` could well be renamed to `LanguageSymEval` or something similar.

To give you a difference, the `LanguageSMT` translation of `if ... then ... else ...` uses the `ifthenelse` primitive in Z3. However, the `LanguageSMTBranches` translation specifies that when an `if` is found, two *new* branches must be created:
1. Add `condition = True` to the list of known things, continue with the `then` term,
2. Add `condition = False` to the list of known things, continue with the `else` term.

Reply:
I won't do that in this PR, I'll do it on the next, which will remove the `Logger` too. I like `LanguageSymEval`!